src/js/crud.js

```diff
@@ -0,0 +1,168 @@
+// Check if authorized
+if (sessionStorage.getItem("auth") !== "granted") {
+  alert("Unauthorized access!");
+  window.location.href = "login.html";
+}
+
+import { fetchFromMultipleBases, getBaseUrls } from './fetch.js';
+
+// Helper for base URL
+async function getBaseUrl(endpoint) {
+  const bases = await getBaseUrls();
+  return `${bases[0]}/${endpoint}`;
+}
+
+// Predefined tags
+const predefinedTags = [
+  "bread", "artisan", "savory", "vegan", "special!", 
+  "pastry", "buttery", "breakfast", "sweet", "dessert",
+  "chocolate", "fruit", "snack", "cake", "citrus"
+];
+
+// Helpers
+function highlightError(inputEl) {
+  inputEl.style.borderColor = "red";
+  inputEl.focus();
```

Comment:
Hier heb je geen optional chaining toegepast. En daarnaast is er geen check of het element wel bestaad.